### PR TITLE
Add support for optional attributes

### PR DESCRIPTION
**Pull request by vogre**
Added a single unittest

### CODE REVIEW

File: src/test_rpc.ts [hunk 1]
```diff
@@ -130,6 +130,37 @@ export default async function test(client: TestClient) {
...
   });
 
+  it('handles multiple parameters', async () => {
+    const schema = `
+export interface Test {
+  bar: {
+    params: {
+      b: string;
+      a: number;
+      c: string | undefined;
```
Comment (bergundy) [inline src/test_rpc.ts]: please test optional params (`c?: string`), this case already works without the fix

File: templates/ts/server.ts [hunk 2]
```diff
@@ -65,7 +65,7 @@ export class {{name}}Router {
       const coerced = coerceWithSchema(this.schemas.{{name}}, args, schema);
       const params = def.properties.{{name}}.properties.params;
       const order = (params as any).propertyOrder || [];
-      const sortedArgs = Object.entries(coerced).sort(([a], [b]) => order.indexOf(a) - order.indexOf(b)).map(([_, v]) => v);
+      const sortedArgs = order.map(param => coerced[param]);
```
Comment (bergundy) [inline templates/ts/server.ts]: Like!

**Review by bergundy — CHANGES_REQUESTED** [review]
Thanks!
Please also change the generated server code.

**Comment by vogre** [comment]
@bergundy : PTAL, also changed the server code.

File: src/test/test_rpc.ts [hunk 3]
```diff
@@ -127,6 +127,36 @@ export default async function test(client: TestClient) {
...
+  public async bar(b: string, a: number, c?: string, d?: string): Promise<string> {
+    return d ? \`\${d} \${b} \${a}\` : \`\${a}\`;
+  }
+}
+`;
+    const tester = `
+import { TestClient } from './client';
+export default async function test(client: TestClient) {
+ expect(await client.bar('hello', 3, undefined, 'x')).to.equal('x hello 3');
+ expect(await client.bar('hello', 3, undefined, undefined)).to.equal('3');
```
Comment (bergundy) [inline src/test/test_rpc.ts]: Could have been tested with parameters omitted.

**Review by bergundy — APPROVED** [review]
Thanks!

**Pull request merged**